Services/GameStateService.cs

```diff
@@ -41,5 +53,80 @@ public PokerTable LoadGameStateFromFile(string filePath)
+            //Console.WriteLine(json);
+
+            // Step 2: Deserialize JSON to PokerTable Data
+            PokerTable pokerTable = JsonSerializer.Deserialize<PokerTable>(json, _jsonSerializerOptions);
+
+            // Step 3: Process PokerTable data
+            int entryStack = FindStartingStack(pokerTable);
+            var (playersAboveEntryStack, playersBehindEntryStack) = CategorizePlayers(pokerTable, entryStack);
+
+            return new LoadGameResponse
+            {
+                HighestStack = FindHighestStack(pokerTable),
+                LowestStack = FindLowestStack(pokerTable),
+                EntryStack = entryStack,
+                ListOfPlayersAboveEntryStack = playersAboveEntryStack,
+                ListOfPlayersBehindEntryStack = playersBehindEntryStack,
+                SuperFolksCount = FindSuperFolksCount(pokerTable)
+            };
+
+        }
+        private int FindStartingStack(PokerTable pokerTable)
+        {
+            int totalStack = pokerTable.Players.Sum(player => player.ChipCount);
+            int startingStack = totalStack / pokerTable.Players.Count();
+            return startingStack;
+        }
+
+        private PlayerStackInfo FindHighestStack(PokerTable pokerTable)
+        {
+            return pokerTable.Players
+                .Select(player => new PlayerStackInfo { Name = player.Name, Value = player.ChipCount })
+                .OrderByDescending(info => info.Value)
+                .FirstOrDefault();
+        }
+
+        private PlayerStackInfo FindLowestStack(PokerTable pokerTable)
+        {
+            return pokerTable.Players
+                .Select(player => new PlayerStackInfo { Name = player.Name, Value = player.ChipCount })
+                .OrderBy(info => info.Value)
+                .FirstOrDefault();
+        }
+
+        private (List<string>, List<string>) CategorizePlayers(PokerTable pokerTable, int entryStack)
+        {
+            var playersAbove = pokerTable.Players
+                .Where(player => player.ChipCount > entryStack)
+                .Select(player => player.Name)
+                .ToList();
+
+            var playerBelow = pokerTable.Players
+                .Where(player => player.ChipCount <= entryStack)
+                .Select(player => player.Name)
+                .ToList();
+
+            return(playersAbove, playerBelow);
+        }
+
+        //superFolks are Players that plays a hand of 2,7 offsuit
+        private int FindSuperFolksCount(PokerTable pokerTable)
+        {
+            var superFolksCount = pokerTable.Players.Count(player =>
+            player?.CurrentHand?.Card1 != null && player?.CurrentHand?.Card2 != null &&
+            ((player.CurrentHand.Card1.Rank == Rank.Two && player.CurrentHand.Card2.Rank == Rank.Seven) || (player.CurrentHand.Card1.Rank == Rank.Seven && player.CurrentHand.Card2.Rank == Rank.Two)) && (player.CurrentHand.Card1.Suit != player.CurrentHand.Card2.Suit));
```

Comment:
ten warunek ` player?.CurrentHand?.Card1 != null && player?.CurrentHand?.Card2 != null`
można całkowicie usunąć jeśli w linijce niżej dodasz znaki zapytania:

np tak:

`player?.CurrentHand?.Card1?.Rank == Rank.Two`

Reply:
tu musiałem wszędzie dawać te znaki zapytania, żeby mi się skompilowało i żeby potem nie wywaliło błędu "Object reference not set to an instance of an object.".
`var superFolksCount = pokerTable.Players
    .Count(player => ((player?.CurrentHand?.Card1?.Rank == Rank.Two && player?.CurrentHand?.Card2?.Rank == Rank.Seven)
    || (player?.CurrentHand?.Card1?.Rank == Rank.Seven && player?.CurrentHand?.Card2?.Rank == Rank.Two))
    && (player?.CurrentHand?.Card1?.Suit != player?.CurrentHand?.Card2?.Suit)
    && player?.CurrentHand?.Card2?.Suit != null);`
to kod z nowego commita - nie wiem jak się do niego odnieść tutaj

Reply:
Tutaj użyłeś tych utility co nie? Dzięki temu kod się skróci 3 razy i będzie 5x bardziej czytelny

Reply:
tak, w ostatnim commicie to już dodałem:
```
private int FindSuperFolksCount(PokerTable pokerTable)
{
    var superFolksCount = pokerTable.Players
        .Count(player => !player.CurrentHand.IsSuited() && player.CurrentHand.HasSpecificRanks((Rank)2, (Rank)7));

    return superFolksCount;
}
```